clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -940,6 +948,204 @@ public void testUnsupportedForMessageFormatInTxnOffsetCommit() {
         assertFatalError(UnsupportedForMessageFormatException.class);
     }
 
+    @Test
+    public void testFencedInstanceIdInTxnOffsetCommitByGroupMetadata() {
```

Comment:
The 6 tests here are primarily evaluating that when we are on groupMetadata mode, we could correctly detect `FENCED_INSTANCE_ID`, `UNKNOWN_MEMBER_ID` and `ILLEGAL_GENERATION` exceptions. If we are using old API and encounter these exceptions, that would be an illegal state.